### PR TITLE
suggestions for users

### LINKED ISSUE #6273: Ohjaajan kurssisuosituksen näyttäminen etusivulla
Jos opiskelija ei ole millään kurssilla ja ohjaaja on suositellut seuraavaksi jotakin kurssia, etusivulta voi ilmoittautua uudelle kurssille suoraan.

### PR DESCRIPTION
**Pull request by avukkonen**
Adds suggestions for students in the front page after sign in

Closes #6273

### CODE REVIEW

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/index/body/continue-studies-panel.tsx [hunk 1]
```diff
@@ -1,10 +1,19 @@
...
+import { connect, Dispatch } from "react-redux";
+import { AnyActionType } from "~/actions";
+import { bindActionCreators } from "redux";
 import { i18nType } from "~/reducers/base/i18n";
 import { StatusType } from "~/reducers/base/status";
 import { WorkspaceMaterialReferenceType } from "~/reducers/workspaces";
 import { StateType } from "~/reducers";
+import { useCourseCarousel } from "~/components/general/carousel/hooks/use-course-carousel";
+import WorkspaceSignup from "~/components/coursepicker/dialogs/workspace-signup";
+import Link from "../../general/link";
```
Comment (TeroOtavia) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/index/body/continue-studies-panel.tsx:15]: Unused imports

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/index/body/continue-studies-panel.tsx [hunk 2]
```diff
@@ -1,10 +1,19 @@
...
+import Button from "~/components/general/button";
```
Comment (TeroOtavia) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/index/body/continue-studies-panel.tsx:16]: Unused imports

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/index/body/continue-studies-panel.tsx [hunk 3]
```diff
@@ -1,10 +1,19 @@
-import Link from "../../general/link";
 import * as React from "react";
-import { connect } from "react-redux";
+import {
+  displayNotification,
+  DisplayNotificationTriggerType,
+} from "~/actions/base/notifications";
```
Comment (TeroOtavia) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/index/body/continue-studies-panel.tsx:5]: These are not used anywhere. Remove imports and props and mapDispatchTopProps definings

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/index/body/next-studies-panel.tsx [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+ * @returns JSX.Element
+ */
+const NextStudiesPanel: React.FC<NextStudiesPanelProps> = (props) => {
+  const { courseCarousel } = useCourseCarousel(
+    props.status.userSchoolDataIdentifier,
+    props.status.userId,
+    props.status.isStudent,
+    props.displayNotification
+  );
+
```
Comment (TeroOtavia) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/index/body/next-studies-panel.tsx:41]: You are passing more parameters than function can hold. There is no boolean parameter to check if user is student. If you have changed useCourseCarousel custom hook, then changes are not shown correctly

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/index/body/next-studies-panel.tsx [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+  lastWorkspace: WorkspaceMaterialReferenceType;
+  displayNotification: DisplayNotificationTriggerType;
+}
+
+/**
+ * NextStudiesPanel
+ * @param props NextStudiesPanelProps
+ * @returns JSX.Element
+ */
+const NextStudiesPanel: React.FC<NextStudiesPanelProps> = (props) => {
```
Comment (TeroOtavia) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/index/body/next-studies-panel.tsx:34]: Is it necessary to show all suggestedWorkspace here? Wouldn't student counselors suggestion be enough. Like suggested for next and/or suggested for optional?

As now we are loading same big chunk of data for two places, for Studies view and now for this. If we are ruling that enviroment frontpage only gives sneak peaks or most important bits of data, then I would say this is overkill.

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/index/body/next-studies-panel.tsx [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+import "~/sass/elements/panel.scss";
+import "~/sass/elements/item-list.scss";
+
+/**
+ * NextStudiesPanelProps
+ */
+interface NextStudiesPanelProps {
+  i18n: i18nType;
+  status: StatusType;
+  lastWorkspace: WorkspaceMaterialReferenceType;
```
Comment (TeroOtavia) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/index/body/next-studies-panel.tsx:25]: Is not used anywhere, remove prop and mapStateToProps definings

File: muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/index/body/next-studies-panel.tsx [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+  displayNotification,
+  DisplayNotificationTriggerType,
+} from "~/actions/base/notifications";
+import { connect, Dispatch } from "react-redux";
+import { AnyActionType } from "~/actions";
+import { bindActionCreators } from "redux";
+import { i18nType } from "~/reducers/base/i18n";
+import { StatusType } from "~/reducers/base/status";
+import { StateType } from "~/reducers";
+import { useCourseCarousel } from "~/components/general/carousel/hooks/use-course-carousel";
```
Comment (TeroOtavia) [inline muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/index/body/next-studies-panel.tsx:12]: Unused import

**Review by TeroOtavia — CHANGES_REQUESTED** [review]
Fix imports and then its ready to merge

**Review by jjlankinen — DISMISSED** [review]
Tested manually UI and feature wise and works as it should. 

Possible other code reviews still needed to be apporved before merge.

**Review by TeroOtavia — DISMISSED** [review]

**Review by jjlankinen — APPROVED** [review]
Approved twice.

**Pull request merged**